contribute/style-guides/testing.md

```diff
@@ -0,0 +1,251 @@
+# Testing Guidelines
+
+The goal of this document is to address the most frequently asked "How to" questions related to unit testing.
+
+## Testing User Interactions
+
+We use the [user-event](https://testing-library.com/docs/user-event/intro) library for simulating user interactions during testing. This library is preferred over the built-in `fireEvent` method, as it more accurately mirrors real user interactions with elements.
+
+There are two important considerations when working with `userEvent`:
+
+1. All methods in `userEvent` are asynchronous, and thus require the use of `await` when called.
+2. Directly calling methods from `userEvent` may not be supported in future versions. As such, it's necessary to first call `userEvent.setup()` prior to the tests. This method returns a `userEvent` instance, complete with all its methods. This setup process can be simplified using a utility function:
+
+```tsx
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+
+function setup(jsx: JSX.Element) {
+  return {
+    user: userEvent.setup(),
+    ...render(jsx),
+  };
+}
+
+it('should render', async () => {
+  const { user } = setup(<Button />);
+  await user.click(screen.getByRole('button'));
+});
+```
+
+## Debugging Tests
+
+There are a few utilities that can be useful for debugging tests:
+
+- [screen.debug()](https://testing-library.com/docs/queries/about/#screendebug) - This function prints a human-readable representation of the document's DOM tree when called without arguments, or the DOM tree of specific node(s) when provided with arguments. It is internally using `console.log` to log the output to terminal.
+- [Testing Playground](https://testing-playground.com/) - An interactive sandbox that allows testing which queries work with specific HTML elements.
+- [logRoles](https://testing-library.com/docs/dom-testing-library/api-debugging/#prettydom) - A utility function that prints out all the implicit ARIA roles for a given DOM tree.
+
+## Testing Select Components
+
+Here, the [OrgRolePicker](https://github.com/grafana/grafana/blob/main/public/app/features/admin/OrgRolePicker.tsx) component is used as an example. This component essentially serves as a wrapper for the `Select` component, complete with its own set of options.
```

Comment:
instead of linking to main, it might be better to link to a specific commit like https://github.com/grafana/grafana/blob/38863844e7ac72c7756038a1097f89632f9985ff/public/app/features/admin/OrgRolePicker.tsx#L18 ?

Reply:
Yeah, good point 👍